physx/src/controller.rs

```diff
@@ -0,0 +1,93 @@
+#![warn(clippy::all)]
+#![warn(rust_2018_idioms)]
+
+use crate::transform::gl_to_px_v3;
+use glam::Vec3;
+use physx_sys::PxControllerFilterCallback;
+use physx_sys::{
+    PxController, PxControllerFilters, PxControllerFilters_new, PxController_getActor,
+    PxController_getPosition, PxController_move_mut, PxController_release_mut,
+    PxController_setPosition_mut, PxExtendedVec3, PxQueryFilterCallback, PxRigidDynamic,
+};
+use std::ptr::null;
+
+pub struct Controller {
+    controller: *mut PxController,
+    filters: PxControllerFilters,
+}
+
+impl Controller {
```

Comment:
Can we use the `[physx_type]` macro pattern here? It's not straight forward to combine it with data ownership, unfortunately. Maybe it'd make sense to separate this into two parts, a wrapper around the PhysX-sys API (the trait) and the actual controller? It feels like the combination of these two data pieces might belong more at the application level. (I.e., in my specific character implementation for my game). 

So something like:

```rust
#[px_type]
impl Controller {
    // all API functions for PxController 
}

struct Controllable {
    controller: Controller,
    filters: ControllerFilters,
}
```

Reply:
Ok, I wasn't quite sure what that macro did. I think I get it now. 

Regarding Controller/Controllable, would it make sense to use Deref and DerefMut? 
```impl Deref for Controllable {
    type Target = Controller;
    fn deref(&self) -> &Controller {
        &self.controller
    }
}

impl DerefMut for Controllable {
    fn deref_mut(&mut self) -> &mut Controller {
        &mut self.controller
    }
}
```

Reply:
No, but I'm not sure if such a `Controllable` belongs in physx-rs at all since it's not a PxSomething (right?). We only use Deref/DerefMut to emulate direct inherited function calls since there is no mechanic to do that nicely in Rust. In this case, there is nothing called a PxControllable that _is-a_ PxController as far as I know. But the `physx_type` macro can do it for the subtypes of `PxController`: 

```rust
#[physx_type(inherit = "Controller")]
impl CapsuleController {
    // PxCapsuleControllerAPI
};
```

This'll implement appropriate `GetRaw`, `Deref` and `DerefMut` for `PxType<PxCapsuleController>` and alias it as CapsuleController.